pkg/trace/api/api_test.go

```diff
@@ -368,6 +368,31 @@ func TestReceiverDecodingError(t *testing.T) {
 	})
 }
 
+func TestReceiverUnexpectedEOF(t *testing.T) {
+	assert := assert.New(t)
+	conf := newTestReceiverConfig()
+	r := newTestReceiverFromConfig(conf)
+	server := httptest.NewServer(http.HandlerFunc(r.handleWithVersion(v05, r.handleTraces)))
+	// we get to read the header and the entire dictionary, but the request claims to be much larger
```

Comment:
Not sure it's clear what the first 3 request bytes are meant to be from the description.